src/main/java/pl/trainingapp/repositories/ExerciseRepository.java

```diff
@@ -0,0 +1,38 @@
+package pl.trainingapp.repositories;
+
+import org.springframework.stereotype.Repository;
+import pl.trainingapp.entities.Exercise;
+import pl.trainingapp.exceptions.EntityNotFoundException;
+
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+public class ExerciseRepository {
+    
+    public void addExercise(String name) {
+        int maxId = ExerciseDatabase.exercises.stream().mapToInt(Exercise::getId).max().orElse(1);
+
+        ExerciseDatabase.exercises.add(new Exercise(maxId, name));
+    }
+
+    public void deleteExerciseById(int id) throws EntityNotFoundException {
+        Optional<Exercise> exercise = Optional.ofNullable(ExerciseDatabase
+                .exercises.stream().filter(e -> e.getId() == id)
+                .findFirst().stream().findFirst().orElseThrow(() -> new EntityNotFoundException("Cannot find entity in database.")));
```

Comment:
Optional.ofNullable is unnecessary, findFirst will already return Optional. Also you have stream().findFirst() two times. I think here you don't really need to throw entity not found. If we will try to delete entity that doesn't exist then operation was succesful - no entity. So you can do:

```
Optional<Exercise> exercise = ExerciseDatabase
                .exercises.stream().filter(e -> e.getId() == id)
                .findFirst();
                
if (excercise.isPresent()) {
  ExerciseDatabase.exercises.remove(exercise.get());
}

Reply:
ok